File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        nums.sort()

        map = {}
        for num in nums:
            map[num] = 0   # just used to dedupe keys

        arr = []
        for val in map.keys():
            arr.append(val)

        # Now arr is sorted + unique
        best = 1
        curr = 1

        res = []
        for i in range(len(arr) - 1):
            a, b = arr[i], arr[i+1]
            if a + 1 == b:
                curr += 1
            else:
                best = max(best, curr)
                curr = 1

        best = max(best, curr)
        return best

File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        int j = 0;
        for(int i = j+1; i < nums.length && j < nums.length; i++){
            System.out.printf("%d %d\n", j, i);
            if(nums[j] + nums[i] == target){
                result[0] = j;
                result[1] = i;
                return result;
            }
            else if(i == nums.length - 1){
                j++;
                i = j;
            }
        }
        return result;
    }
}
